pygmt/src/tilemap.py

```diff
@@ -0,0 +1,155 @@
+"""
+tilemap - Plot XYZ tile maps.
+"""
+from pygmt.clib import Session
+from pygmt.datasets.tile_map import load_tile_map
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+try:
+    import rioxarray
+except ImportError:
+    rioxarray = None
+
+
+@fmt_docstring
+@use_alias(
+    B="frame",
+    E="dpi",
+    I="shading",
+    J="projection",
+    M="monochrome",
+    N="no_clip",
+    Q="nan_transparent",
+    # R="region",
+    V="verbose",
+    c="panel",
+    p="perspective",
+    t="transparency",
+)
+@kwargs_to_strings(c="sequence_comma", p="sequence")  # R="sequence",
+def tilemap(
+    self, region, zoom="auto", source=None, lonlat=True, wait=0, max_retries=2, **kwargs
+):
+    r"""
+    Plots an XYZ tile map.
+
+    This function loads XYZ tile maps from a tile server or local file using
+    :func:`pygmt.datasets.load_tile_map` into a georeferenced form, and plots
+    the tiles as a basemap or overlay using :meth:`pygmt.Figure.grdimage`.
+
+    **Note**: By default, standard web map tiles served in a Web Mercator
```

Comment:
```suggestion
    **Note**: By default, standard web map tiles served in a Spherical Mercator
```
Although the terms are meaning the same, I think we should only use one definition throughout the whole documentation to avoid confusion. I would prefer Spherical Mercator instead of Web Mercator.

Reply:
Agree, and https://epsg.io/3857 uses Spherical Mercator too.